src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -410,95 +433,180 @@ protected override bool ReceivePluginInternal(object message)
                 case ReplayTaggedMessages replay:
                     ReplayTaggedMessagesAsync(replay)
                         .PipeTo(replay.ReplyTo, success: h => new RecoverySuccess(h), failure: e => new ReplayMessagesFailure(e));
-                    break;
+                    return true;
+                case ReplayAllEvents replay:
+                    ReplayAllEventsAsync(replay)
+                        .PipeTo(replay.ReplyTo, success: h => new EventReplaySuccess(h),
+                            failure: e => new EventReplayFailure(e));
+                    return true;
                 case SubscribePersistenceId subscribe:
                     AddPersistenceIdSubscriber(Sender, subscribe.PersistenceId);
                     Context.Watch(Sender);
-                    break;
-                case SubscribeAllPersistenceIds subscribe:
```

Comment:
Are you conflating this and `SubscribeCurrentPersistenceIds`? There's a difference between both of them:

1. Current persistence Ids is a finite query - it terminates once it reaches the end of all "known" persistent ids;
2. All persistence ids is an infinite query - it continues indefinitely and emits new notifications as new events are added.

Why do we only handle one of these inside the journal? Is it because you're using the `request.Offset` to manage this?

Reply:
No, that is an outdated code. `SubscribePersistenceId` and `SelectCurrentPersistenceIds` are handled separately.